pkg/vm/memory/memory_value.go

```diff
@@ -99,40 +100,32 @@ func (address MemoryAddress) String() string {
 	)
 }
 
+// This is an abreviation for simplicity
+type memoryValue = mo.Either[*f.Element, *MemoryAddress]
+
 // Stores all posible types that can be stored in a Memory cell,
 //
 //   - either a Felt value (an `f.Element`),
 //   - or a pointer to another Memory Cell (a `MemoryAddress`)
-//
-// Both members cannot be non-nil at the same time
-type MemoryValue struct {
-	felt    *f.Element
-	address *MemoryAddress
-}
+type MemoryValue memoryValue
```

Comment:
You could directly type:
```
type MemoryValue mo.Either[*f.Element, *MemoryAddress]
```
I think the type `memoryValue` is not used anywhere else

Reply:
I use its generated cast  function